pyomo/contrib/mindtpy/config_options.py

```diff
@@ -1,251 +1,490 @@
+        domain=bool
     ))
-    CONFIG.declare("use_dual_bound", ConfigValue(
-        default=True,
-        description="add dual bound constraint to enforce the objective function should improve on the best found dual bound",
+    CONFIG.declare('use_bb_tree_incumbent', ConfigValue(
+        default=False,
+        description='Whether to use the incumbent solution of branch & bound tree in ROA single tree when regularization problem is infeasible',
         domain=bool
     ))
-    return CONFIG
+    CONFIG.declare('sqp_lag_scaling_coef', ConfigValue(
+        default='fixed',
+        domain=In(['fixed', 'variable_dependent']),
+        description='the coefficient used to scale the L2 norm in sqp_lag'
+    ))
+
+
+
+def check_config(config):
+    # configuration confirmation
+    if config.add_regularization in {'grad_lag', 'hess_lag', 'hess_only_lag', 'sqp_lag'}:
+        config.calculate_dual = True
+    if config.add_regularization is not None:
+        if config.regularization_mip_threads == 0 and config.threads > 0:
+            config.regularization_mip_threads = config.threads
+            config.logger.info('Set regularization_mip_threads equal to threads')
+        if config.single_tree:
+            config.add_cuts_at_incumbent = True
+            # if no method is activated by users, we will use use_bb_tree_incumbent by default
+            if not (config.reduce_level_coef or config.use_bb_tree_incumbent):
+                config.use_bb_tree_incumbent = True
+        if config.mip_regularization_solver is None:
+            config.mip_regularization_solver = config.mip_solver
+    if config.single_tree:
+        config.iteration_limit = 1
+        config.add_slack = False
+        config.mip_solver = 'cplex_persistent'
+        config.logger.info(
+            'Single tree implementation is activated. The defalt MIP solver is cplex_persistent')
+        if config.threads > 1:
+            config.threads = 1
+            config.logger.info(
+                'The threads parameter is corrected to 1 since lazy constraint callback conflicts with multi-threads mode.')
+    # if the slacks fix to zero, just don't add them
+    if config.max_slack == 0.0:
+        config.add_slack = False
+
+    if config.strategy == 'GOA':
+        config.add_slack = False
+        config.use_mcpp = True
+        # config.integer_to_binary = True
+        config.equality_relaxation = False
+        config.use_fbbt = True
+        # add_no_good_cuts is Ture by default in GOA
+        if not config.add_no_good_cuts and not config.use_tabu_list:
+            config.add_no_good_cuts = True
+            config.use_tabu_list = False
+    elif config.strategy == 'FP':  # feasibility pump alone
+        config.init_strategy = 'FP'
+        config.iteration_limit = 0
+    if config.init_strategy == 'FP':
+        # TODO: Choose one from the following two
```

Comment:
Does this TODO need to be resolved before merging?

Reply:
Resolved.